File: src/main/java/kr/co/bookand/backend/issue/domain/IssueImage.java
package kr.co.bookand.backend.issue.domain;

import kr.co.bookand.backend.common.domain.BaseEntity;
import lombok.*;

import javax.persistence.*;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Builder
public class IssueImage extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "issue_image_id")
    private Long id;

    private String imageUrl;

    @ManyToOne(fetch = FetchType.LAZY)
    private Issue issue;
}

File: src/main/kotlin/kr/co/bookand/backend/issue/domain/IssueImage.kt
package kr.co.bookand.backend.issue.domain

import kr.co.bookand.backend.common.domain.BaseEntity
import javax.persistence.*

@Entity
class IssueImage(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "issue_image_id")
    var id: Long = 0,
    var imageUrl: String,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "issue_id")
    var issue: Issue

) : BaseEntity() {

    constructor(imageUrl: String, issue: Issue) : this(
        id = 0,
        imageUrl = imageUrl,
        issue = issue
    )
}